cfgov/unprocessed/js/molecules/GlobalSearch.js

```diff
@@ -0,0 +1,197 @@
+   */
+  function init() {
+    var inputSelector = '.' + BASE_CLASS + '_content-form input';
+    var clearBtnSelector =
+      '.' + BASE_CLASS + ' .input-contains-label_after__clear';
+    var searchIconSelector =
+      '.' + BASE_CLASS + ' .input-contains-label_before__search'
+    var searchBtnSelector = '.' + BASE_CLASS + ' .input-with-btn_btn button';
+
+    _searchInputDom = _contentDom.querySelector( inputSelector );
+    var searchIconDom = _contentDom.querySelector( searchIconSelector );
+    _clearBtnDom = _contentDom.querySelector( clearBtnSelector );
+
+    _triggerDom.addEventListener( 'click', _triggerClicked );
+    _clearBtnDom.addEventListener( 'mousedown', _clearClicked );
+    _searchInputDom.addEventListener( 'keyup', _inputTyped );
+    _contentDom.addEventListener( 'blur', _searchBlurred, true );
+    searchIconDom.addEventListener( 'click', _searchIconClicked );
+
+    _setClearBtnState( _searchInputDom.value );
+
+    return this;
+  }
+
+  /**
+   * Event handler for when the search icon is clicked in the
+   * expanded state at desktop sizes. Closes the search box.
+   */
+  function _searchIconClicked() {
+    _collapseIfDesktop();
+  }
+
+  /**
+   * Event handler for when the search input loses focus.
+   * Closes the search input or dispatches a click if
+   * the target is a button.
+   */
+  function _searchBlurred( event ) {
+    var target = event.target;
+    if ( target.nodeName === 'BUTTON' ) {
+      target.click();
+    } else {
+      _collapseIfDesktop();
+    }
+  }
+
+  /**
+   * Collapse the search box if screen is at desktop sizes.
+   */
+  function _collapseIfDesktop() {
+    var currentBreakpoint = breakpointState.get();
+    if ( ( currentBreakpoint.isBpMED ||
+         currentBreakpoint.isBpLG ||
+         currentBreakpoint.isBpXL ) ) {
+      collapse();
+    }
+  }
+
+  /**
+   * Event handler for when the search input trigger is clicked,
+   * which opens/closes the search input.
+   */
+  function _triggerClicked() {
+    if ( _isExpanded ) {
+      collapse();
+    } else {
+      expand();
+    }
+  }
+
+  /**
+   * Open the search box.
+   * @returns {Object} An GlobalSearch instance.
+   */
+  function expand() {
+    _isExpanded = true;
+    _triggerDom.setAttribute( 'aria-expanded', 'true' );
+    _contentDom.setAttribute( 'aria-expanded', 'true' );
+    _searchInputDom.select();
+
+    return this;
+  }
+
+  /**
+   * Close the search box.
+   * @returns {Object} An GlobalSearch instance.
+   */
+  function collapse() {
+    _isExpanded = false;
+    _triggerDom.setAttribute( 'aria-expanded', 'false' );
+    _contentDom.setAttribute( 'aria-expanded', 'false' );
+
+    return this;
+  }
+
+  /**
+   * @returns {boolean} Whether the search box is expanded or not.
+   *   This can be used to determine if the box is open
+   *   and needs to be closed if an overlapping dialog box is opened.
+   */
+  function isExpanded() {
+    return _isExpanded;
+  }
```

Comment:
That's a good point. I added this in anticipation that the mega menu would need to close the search menu, but you're right it doesn't need to check first. Those checks can and should happen in the expand/collapse methods. Fixed in https://github.com/cfpb/cfgov-refresh/commit/8663c30f65b433a2ccd0703c1f05fc4e9e5796f1